### PR TITLE
[Event Hubs] Processor + Extensions April 2024 Release prep

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to prepare the Event Hubs Processor and Functions extensions packages for the April 2024 release.  Also included are reference version bumps for the April 2024 release of Service Bus.

### NOTE

This cannot be merged until the April release of `Azure.Messaging.EventHubs` and `Azure.Messaging.ServiceBus` have been completed.

### CODE REVIEW

**Review by m-redding — APPROVED** [review]

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -246,8 +246,8 @@
     <PackageReference Update="ApprovalUtilities" Version="3.0.22" />
     <PackageReference Update="Azure.Identity" Version="1.10.4" />
     <PackageReference Update="Azure.Messaging.EventGrid" Version="4.17.0" />
-		<PackageReference Update="Azure.Messaging.EventHubs.Processor" Version="5.11.0" />
-    <PackageReference Update="Azure.Messaging.ServiceBus" Version="7.16.0" />
+		<PackageReference Update="Azure.Messaging.EventHubs.Processor" Version="5.11.2" />
```
Comment (jsquire) [inline eng/Packages.Data.props:249]:
```suggestion
		<PackageReference Update="Azure.Messaging.EventHubs.Processor" Version="5.11.1" />
```

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/api/Microsoft.Azure.WebJobs.Extensions.EventHubs.netstandard2.0.cs [hunk 2]
```diff
@@ -30,6 +30,7 @@ public partial class EventHubOptions : Microsoft.Azure.WebJobs.Hosting.IOptionsF
         public int BatchCheckpointFrequency { get { throw null; } set { } }
         public Azure.Messaging.EventHubs.EventHubsRetryOptions ClientRetryOptions { get { throw null; } set { } }
         public System.Uri CustomEndpointAddress { get { throw null; } set { } }
+        public bool EnableCheckpointing { get { throw null; } set { } }
```
Comment (christothes) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/api/Microsoft.Azure.WebJobs.Extensions.EventHubs.netstandard2.0.cs:33]: Is this API change expected?

Comment (jsquire) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/api/Microsoft.Azure.WebJobs.Extensions.EventHubs.netstandard2.0.cs:33]: Yes.  It didn't get exported earlier due to the CI pipeline being disabled.   It's included here to generate an API View for architect review.

**Review by christothes — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md [hunk 3]
```diff
@@ -1,10 +1,6 @@
 # Release History
 
-## 5.12.0-beta.1 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
+## 5.11.2 (2024-04-09)
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md:3]:
```suggestion
## 5.11.2 (2024-04-10)
```

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/CHANGELOG.md [hunk 4]
```diff
@@ -1,17 +1,11 @@
 # Release History
 
-## 6.3.0-beta.1 (Unreleased)
+## 6.3.0 (2024-04-09)
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/CHANGELOG.md:3]:
```suggestion
## 6.3.0 (2024-04-10)
```

**Pull request merged**